src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestStorage.java

```diff
@@ -250,26 +262,29 @@ synchronized SnowflakeFileTransferMetadataWithAge refreshSnowflakeMetadata(boole
       return fileTransferMetadataWithAge;
     }
 
-    Map<Object, Object> payload = new HashMap<>();
-    payload.put("role", this.role);
-    ConfigureResponse response = this.makeClientConfigureCall(payload);
-
+    ConfigureResponse response = this.configureCallHandler.makeConfigureCall();
     // Do not change the prefix everytime we have to refresh credentials
     if (Utils.isNullOrEmpty(this.clientPrefix)) {
       this.clientPrefix = createClientPrefix(response);
```

Comment:
lets get the clientPrefix from StorageManager interface?

Reply:
and not "create" it here?

Reply:
Moved to `StorageManager` level.